### PR TITLE
sum/dif_of_prod real and complex update

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/complex/detail/arithmetic.hpp [hunk 1]
```diff
@@ -201,6 +201,23 @@ namespace eve::detail
...
   }
 
+  //================================================================================================
+  //  trivial extension of some real tetrary functions
+  //================================================================================================
+  template<typename Z1, typename Z2, typename Z3, typename Z4>
+  EVE_FORCEINLINE auto
+  complex_nary_dispatch(eve::tag::sum_of_prod_, Z1 const& z1, Z2 const& z2, Z3 const& z3, Z4 const& z4) noexcept
+  {
+    return z1*z2 + z3*z4;
```
Comment (jfalcou) [inline include/eve/module/complex/detail/arithmetic.hpp:211]: no FMA ?

Comment (jfalcou) [inline include/eve/module/complex/detail/arithmetic.hpp:211]: no FMA/FNMS ?

File: include/eve/module/core/pedantic/impl/diff_of_prod.hpp [hunk 2]
```diff
@@ -13,32 +13,36 @@
...
                                    , const V& c
                                    , const W& d
-                                   ) noexcept
-{
-  return arithmetic_call(pedantic(diff_of_prod), a, b, c, d);
-}
+                                   ) noexcept -> decltype(a*b+c*d)
+  {
+    using r_t = decltype(a*b+c*d);
+    return pedantic(sum_of_prod)(r_t(a), r_t(b), r_t(c), r_t(d));
```
Comment (jfalcou) [inline include/eve/module/core/pedantic/impl/diff_of_prod.hpp:26]: shouldn't it be `diff_of_prod` ?

**Review by jfalcou — CHANGES_REQUESTED** [review]
-> The raw and pedantic version need test, as the diff_of_prod pedantic is wrong.

File: include/eve/module/complex/detail/arithmetic.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/complex/detail/arithmetic.hpp:211]: ok fma

File: include/eve/module/core/pedantic/impl/diff_of_prod.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/core/pedantic/impl/diff_of_prod.hpp:26]: yes

**Review by jfalcou — APPROVED** [review]

**Pull request merged**